tests/optimize/test_backtesting.py

```diff
@@ -562,6 +564,57 @@ def test_backtest__enter_trade(default_conf, fee, mocker) -> None:
     assert trade
     assert trade.stake_amount == 300.0
 
+
+def test_backtest__enter_trade_futures(default_conf_usdt, fee, mocker) -> None:
+    default_conf_usdt['use_sell_signal'] = False
+    mocker.patch('freqtrade.exchange.Exchange.get_fee', fee)
+    mocker.patch("freqtrade.exchange.Exchange.get_min_pair_stake_amount", return_value=0.00001)
+    mocker.patch("freqtrade.exchange.Exchange.get_max_pair_stake_amount", return_value=float('inf'))
+    patch_exchange(mocker)
+    default_conf_usdt['stake_amount'] = 'unlimited'
+    default_conf_usdt['max_open_trades'] = 2
+    default_conf_usdt['trading_mode'] = 'futures'
+    default_conf_usdt['margin_mode'] = 'isolated'
+    default_conf_usdt['stake_currency'] = 'BUSD'
+    default_conf_usdt['exchange']['pair_whitelist'] = ['.*']
```

Comment:
I'm not sure about this. It seems that despite having 2 `:USDT` pairs in the list - we don't get them selectable for this test.

I suspect it's because of how `is_future()` works for binance (the exchange this is currently testing) - so the only pair "left" is the SOL/BUSD:BUSD one.

I assume the values are changing intentionally - but i've left it failing for now as i'm not sure what your thought behind the comparisons was.

Reply:
> I'm not sure about this. It seems that despite having 2 `:USDT` pairs in the list - we don't get them selectable for this test.

What exactly are you not sure about? What are the :USDT pairs in the list?

> I suspect it's because of how is_future() works for binance (the exchange this is currently testing) - so the only pair "left" is the SOL/BUSD:BUSD one.

You mean `market_is_future`? Which part of it? The part that checks if the type is `swap`/`future` or the part that checks if it's `linear`

> I assume the values are changing intentionally - but i've left it failing for now as i'm not sure what your thought behind the comparisons was.

I don't really know what you're talking about, what values and how are they changing?

When I wrote the test for testing backtesting, I was just trying to check that the `isolated_liq` was being set to what was expected, and I didn't really understand everything else that was going on in the test (because it was already there, and I just wanted to add the liquidation_price part)

Reply:
Nevermind the values - could please double-check the the formula?
i've now updated all values, but the outcome of the formula in the comment is (Totally) different to the real outcome.
maybe the formula is not for binance - or it's wrong (or implemented wrong) - all i know is that the result is ... different.

Reply:
> Nevermind the pairs - could please double-check the the formula? i've now updated all values, but the outcome of the formula in the comment is (Totally) different to the real outcome. maybe the formula is not for binance - or it's wrong (or implemented wrong) - all i know is that the result is ... different.

The amount was just missing a `0` at the end

![](https://i.imgur.com/FTocRrK.png)

Reply:
so what you're saying is - i've been too stupid to copy a value from the debug window into the editor?

fine ... works for me :laughing:

Reply:
> except that there's still differences in the 5th decimal (you get 0.000807 - while the calculation yields 0.0008176). While the difference is slim - it might point to a problem (it's at least something to be suspicious about).

I didn't notice that, I'm checking it right now, should hopefully have an answer in 20 minutes

Reply:
no rush - i also only noticed accidentally - doublechecking if i'm totally mad
your results are quite close though - so it's for sure just a minor thing (my results were WAY off)

Reply:
> no rush - i also only noticed accidentally - doublechecking if i'm totally mad your results are quite close though - so it's for sure just a minor thing (my results were WAY off)

The difference is because of the liquidation buffer, I updated the comments

Reply:
I also added the formula for the freqtrade liquidation price to the docs

Reply:
Great spot! :+1: